magento/magento2-page-builder
# Ensure children do not become orphans

### Description
There is a bug in PageBuilder where child content nodes are going missing. These exist and work just fine when the content is first created, and saved, and shows correctly on the front-end. However, when the form is re-rendered in the admin, the children have been deleted, so saving the page/content a second time deletes these content nodes.

This is because the list of child nodes is replaced rather than added to when walking the node tree.

### Manual testing scenarios
1. Create a new content type with a master template which has several siblings HTML nodes. For example,

```html
<section class="pagebuilder-windmill" attr="data.main.attributes" ko-style="data.main.style" css="data.main.css">
  <div class="wrapper">
    <render args="masterTemplate" ></render>
  </div>

  <div class="pagination" if="data.main.attributes()['data-show-dots']"></div>

  <if args="data.main.attributes()['data-show-arrows']">
    <div class="button-prev"></div>
    <div class="button-next"></div>
  </if>

  <div class="scrollbar" if="data.main.attributes()['data-show-scrollbar']"></div>
</section>
```

2. Create a new Page in the admin and use this new content type.
3. Populate the content type with some child nodes (which will show up in the preview & on the front-end where we have the `<render />` tag)
4. Save (and close) the page in the admin.
5. Check all is well on the front-end.
6. Click "edit" the same page in the admin, make no changes, click "save".
7. Observe the content has been erased by viewing the front-end and/or the admin.

### Checklist
